orchagent/routeorch.cpp

```diff
@@ -1577,3 +1721,76 @@ bool RouteOrch::removeRoutePost(const RouteBulkContext& ctx)
 
     return true;
 }
+
+bool RouteOrch::createRemoteVtep(sai_object_id_t vrf_id, const NextHopKey &nextHop)
+{
+    SWSS_LOG_ENTER();
+    EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
+    VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+    bool status = false;
+    int ip_refcnt = 0;
+
+    status = tunnel_orch->addTunnelUser(nextHop.ip_address.to_string(), nextHop.vni, 0, TUNNEL_USER_IP, vrf_id);
+
+    auto vtep_ptr = evpn_orch->getEVPNVtep();
+    if (vtep_ptr)
+    {
+        ip_refcnt = vtep_ptr->getDipTunnelIPRefCnt(nextHop.ip_address.to_string());
+    }
+    SWSS_LOG_NOTICE("Routeorch Add Remote VTEP %s, VNI %d, VR_ID %lx, IP ref_cnt %d",
+            nextHop.ip_address.to_string().c_str(), nextHop.vni, vrf_id, ip_refcnt);
+    return status;
+}
+
+bool RouteOrch::deleteRemoteVtep(sai_object_id_t vrf_id, const NextHopKey &nextHop)
+{
+    SWSS_LOG_ENTER();
+    EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
+    VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+    bool status = false;
+    int ip_refcnt = 0;
+
+    status = tunnel_orch->delTunnelUser(nextHop.ip_address.to_string(), nextHop.vni, 0, TUNNEL_USER_IP, vrf_id);
+
+    auto vtep_ptr = evpn_orch->getEVPNVtep();
+    if (vtep_ptr)
+    {
+        ip_refcnt = vtep_ptr->getDipTunnelIPRefCnt(nextHop.ip_address.to_string());
+    }
+
+    SWSS_LOG_NOTICE("Routeorch Del Remote VTEP %s, VNI %d, VR_ID %lx, IP ref_cnt %d",
+            nextHop.ip_address.to_string().c_str(), nextHop.vni, vrf_id, ip_refcnt);
+    return status;
+}
+
+bool RouteOrch::removeOverlayNextHops(sai_object_id_t vrf_id, const NextHopGroupKey &ol_nextHops)
+{
+    SWSS_LOG_ENTER();
+    bool status = false;
+
+    SWSS_LOG_NOTICE("Remove overlay Nexthop %s", ol_nextHops.to_string().c_str());
+    for (auto &tunnel_nh : ol_nextHops.getNextHops())
+    {
+        if (!m_neighOrch->getNextHopRefCount(tunnel_nh))
+        {
+            if(!m_neighOrch->removeTunnelNextHop(tunnel_nh))
+            {
+                SWSS_LOG_ERROR("Tunnel Nexthop %s delete failed", ol_nextHops.to_string().c_str());
+            }
+            else
+            {
+                m_neighOrch->removeOverlayNextHop(tunnel_nh);
+                SWSS_LOG_NOTICE("Tunnel Nexthop %s delete success", ol_nextHops.to_string().c_str());
+                SWSS_LOG_NOTICE("delete remote vtep %s", tunnel_nh.to_string(true).c_str());
+                status = deleteRemoteVtep(vrf_id, tunnel_nh);
+                if (status == false)
+                {
+                    SWSS_LOG_NOTICE("Failed to delete remote vtep %s ecmp", tunnel_nh.to_string(true).c_str());
```

Comment:
error log and return false? Please check the error status in the caller methods as well

Reply:
ok. Will do.